### PR TITLE
PHOENIX-6758 During HBase 2 upgrade Phoenix Self healing task fails to create server side connection before reading SYSTEM.TASK

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by virajjasani** [comment]
All tests passed on the build https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1477/1/testReport/

File: phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import org.apache.hadoop.hbase.regionserver.NoSuchColumnFamilyException;
+import org.apache.hadoop.hbase.shaded.protobuf.ProtobufUtil;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.ZooKeeperProtos;
+import org.apache.hadoop.hbase.zookeeper.ZKUtil;
+import org.apache.hadoop.hbase.zookeeper.ZKWatcher;
+import org.apache.hadoop.hbase.zookeeper.ZNodePaths;
+import org.apache.zookeeper.KeeperException;
+
+/**
+ * Admin utility class specifically useful for running Admin APIs in the middle of the
```
Comment (apurtell) [inline phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java:21]: Good comment

File: phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  private static final Logger LOG = LoggerFactory.getLogger(AdminUtilWithFallback.class);
+
+  public static boolean tableExists(Admin admin, TableName tableName)
+      throws IOException, InterruptedException {
+    try {
+      return admin.tableExists(tableName);
+    } catch (IOException e) {
+      if (e instanceof NoSuchColumnFamilyException || (e.getCause() != null
+          && e.getCause() instanceof NoSuchColumnFamilyException)) {
+        LOG.error("Admin API to retrieve table existence failed due to missing CF in meta."
```
Comment (apurtell) [inline phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java:39]: Should only be WARN level because we have a fallback

**Review by apurtell — APPROVED** [review]

File: phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java (hunk 2 above)
Comment (virajjasani) [reply phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java:39]: I can make it WARN level. Besides, this log would appear only once per CQSI initialization. Once initiated, connections are anyways cached.

**Review by kadirozde — APPROVED** [review]
+1, Thanks

**Review by gjacoby126 — APPROVED** [review]
+1

File: phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
+package org.apache.phoenix.query;
```
Comment (gjacoby126) [inline phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java:1]: @virajjasani - this needs an Apache license

**Review by gjacoby126 — CHANGES_REQUESTED** [review]
Looks good but please add a license on AdminUtilWithFallback.java

File: phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java (hunk 3 above)
Comment (virajjasani) [reply phoenix-core/src/main/java/org/apache/phoenix/query/AdminUtilWithFallback.java:1]: Oh what a miss :(
Thanks @gjacoby126

**Review by gjacoby126 — APPROVED** [review]
+1

**Comment by virajjasani** [comment]
Thanks for the reviews @gjacoby126 @kadirozde @apurtell !!! Merged into master and 5.1 branches.

**Pull request merged**